notation/task.json

```
@@ -0,0 +1,204 @@
+            "required": true,
+            "options": {
+                "install": "Install",
+                "sign": "Sign",
+                "verify": "Verify"
+            },
+            "groupName": "basicConfig"
+        },
+        {
+            "name": "artifactRefs",
+            "type": "string",
+            "label": "Artifact references",
+            "defaultValue": "",
+            "required": false,
+            "visibleRule": "command = verify || command = sign",
+            "helpMarkDown": "Container artifact references for signing. If it was not specified, use the artifact reference from previous Docker push task. Example: <registry name>/<repository name>@<digest> If multiple, separate by comma.",
+            "groupName": "commandConfig"
+        },
+        {
+            "name": "signatureFormat",
+            "type": "pickList",
+            "label": "Signature Format",
+            "defaultValue": "cose",
+            "required": false,
+            "options": {
+                "cose": "cose",
+                "jws": "jws"
+            },
+            "visibleRule": "command = sign",
+            "helpMarkDown": "Signature envelope format, options: \"jws\", \"cose\" (default \"jws\")",
+            "groupName": "advancedConfig"
+        },
+        {
+            "name": "allowReferrersAPI",
+            "type": "boolean",
+            "label": "[Experimental] Allow Referrers API",
+            "defaultValue": false,
+            "required": false,
+            "visibleRule": "command = sign || command = verify",
+            "helpMarkDown": "Use the Referrers API to sign signatures, if not supported (returns 404), fallback to the Referrers tag schema",
+            "groupName": "advancedConfig"
+        },
+        {
+            "name": "plugin",
+            "type": "pickList",
+            "label": "Plugin",
+            "defaultValue": "azurekv",
+            "required": true,
+            "options": {
+                "azurekv": "Azure Key Vault Plugin"
+            },
+            "groupName": "pluginConfig"
+        },
+        {
+            "name": "azurekvServiceConection",
+            "type": "connectedService:AzureRM",
+            "label": "Azure Key Vault service connection",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Select the Azure subscription for the key vault if prefer to use service connection for authentication.",
+            "visibleRule": "plugin = azurekv",
+            "groupName": "pluginConfig"
+        },
+        {
+            "name": "keyid",
+            "type": "string",
+            "label": "Key ID",
+            "required": true,
+            "defaultValue": "",
+            "helpMarkDown": "The Key ID is the key or certificate identifier for Azure Key Vault.",
+            "visibleRule": "plugin = azurekv",
+            "groupName": "pluginConfig"
+        },
+        {
+            "name": "cacerts",
```

this name looks a little bit weird, how about 
```suggestion
            "name": "caCertBundle",
```
or something similar

---

Because the flag is named as `cacerts`, do we need to change it to be a different name? Customer may not know what is cert bundle?

---

`cert bundle` is the correct term here, because essentially, this input is a cert bundle :) 
`cacerts` sounds like I can put my certs in any order.

---

How about we keep a consistent naming with the AKV plugin [ca_certs](https://github.com/Azure/notation-azure-kv/blob/main/docs/plugin-config.md#ca_certs) flag for this option but add an informational icon to comment that please input a cert bundle?

---

@FeynmanZhou 
1. As I discussed with @JeyJeyGao, this editor is not a MUST when creating an ADO task pipeline. In other words, a user could use a keyboard to create a pipeline from scratch without any of the information mentioned in the editor. So we cannot rely on the description of the editor during our design. We should make notation ado task's command inputs as clear as possible.
2. As mentioned in our discussion group, users of ADO task does not HAVE to know how azure-kv plugin works. ADO task is a wrapper of (notation CLI + azure-kv plugin). Details are hidden on purpose. Given above, since field `cacerts` works if and only if a ca cert bundle is used, we should rename the field to `caCertBundle`.

---

I see what you mean.
It has been updated to use the `caCertBundle`.
However, there is another issue. For example, if the user uses the Notation CLI + azure-kv on the command line with the ca_certs plugin config, and then moves to the ADO task, the user may become confused about the different naming. Therefore, we need clear documentation to explain this.

@shizhMSFT @FeynmanZhou @yizha1 Are there any concerns?